ckanext/security/authenticator.py

```diff
@@ -101,19 +101,19 @@ def authenticate(self, environ, identity):
         if throttle.is_locked():
             return None
 
-        if auth_user_name is None:
+        if ckan_auth_result is None:
             # Increment the throttle counter if the login failed.
             throttle.increment()
 
         # if the CKAN authenticator has successfully authenticated
         # the request and the user wasn't locked out above,
         # then check the TOTP parameter to see if it is valid
-        if auth_user_name is not None:
-            totp_success = self.authenticate_totp(environ, auth_user_name)
```

Comment:
@ThrawnCA since the underlying CKAN authenticate function no longer returns the username, I've updated to use the username which is already in scope in our implementation.